js/reducers/types.js

```diff
@@ -0,0 +1,18 @@
+// @flow
+
+import { type NavigationState } from 'react-navigation'
+
+import { type AppState } from './appState'
+import { type UserState } from './user'
+
+export type NetworkState = {
+  isConnected: boolean,
+  actionQueue: Array<*>
```

Comment:
is there a more specific super type for all actions?

Reply:
I think we can type actionQueue like this:

```
actionQueue: Array<QueueableAction>
````

where QueueableAction is

```
type QueueableAction = {
  type: string,
  payload?: any,
  meta: {
    retry?: boolean, 
    dismiss?: Array<string>
  }
}
```

Reply:
@cloudify   What you think?

Reply:
better! another option would be to have `QueueableAction` equal to the union of all the actions defined in the app, would that work? (do we have actions coming from other packages?)

Reply:
We have NavigationAction coming from react-navigation, but we already have a generic Action type that is the union of all actions (also NavigationAction) that can be dispatched, so we can use Array<Action> i think. With the migration to redux-saga we'll also have less problems because we'll exclude redux-thunk and all actions will be plain objects (no more function).

Reply:
yes if `Action` is the super type of all possible actions, then `QueueableAction === Action`, so `Array<Action>` is perfect 👍